tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -90,10 +88,17 @@ int IMessagePackFormatter<TraceChunkModel>.Serialize(ref byte[] bytes, int offse
         // overload of IMessagePackFormatter<TraceChunkModel>.Serialize() with `in` modifier on `TraceChunkModel` parameter
         public int Serialize(ref byte[] bytes, int offset, in TraceChunkModel traceChunk, IFormatterResolver formatterResolver, int? maxSize = null)
         {
+            if (traceChunk.SpanCount > 0)
```

Comment:
Better to be safe since an empty chunk will because the next line to fail. I think we intentionally serialize an empty one to ping the agent during startup, for example.